packages/block-editor/src/components/rich-text/index.js

```diff
@@ -323,6 +323,17 @@ export function RichTextWrapper(
 		anchorRef.current?.focus();
 	}
 
+	const onReplaceCallback = shouldDisableEditing
+		? ( blocks ) => {
+				// Find block that does not have the clientId
+				const blockToFocus = blocks.find(
+					( block ) => block.clientId !== clientId
+				);
+
+				// selectBlock( blockToFocus.clientId );
+		  }
+		: onReplace;
```

Comment:
block-list/block.js already conditionally omits/changes onReplace. Could we move this additional condition there?

Reply:
So it turns out we don't need to redefine the `onReplace` — we can just [circumvent the Rich Text's split logic](https://github.com/WordPress/gutenberg/pull/58958/commits/9b2592eabca87d9480a371ab1ce249666c718bb5) if the block is bound.